platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/ConfigurationContextImpl.java

```diff
@@ -147,6 +147,27 @@ public boolean shouldBeVisibleToPlugins() {
     return servicePid != null && configIsNew == null && pidList == null && propertyCount > 0;
   }
 
+  // Contexts can be used in a set
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    ConfigurationContextImpl context = (ConfigurationContextImpl) o;
+
+    return servicePid != null ? servicePid.equals(context.servicePid) : context.servicePid == null;
```

Comment:
✏️  Can be replace with `return Objects.equals(servicePid, context.servicePid);`

Reply:
❓ Can servicePid actually be null?

Reply:
Technically no, according to the contract and where I'm getting it from, it should not be null. But I'm not rolling the dice with our inspection of Felix sources.

Reply:
Then my first comment applies here.